components/experimental/src/dimension/percent/format.rs

```diff
@@ -0,0 +1,302 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use fixed_decimal::{FixedDecimal, Sign};
+
+use crate::alloc::borrow::ToOwned;
+use crate::alloc::string::ToString;
+use alloc::string::String;
+use writeable::Writeable;
+
+use crate::dimension::provider::percent::PercentEssentialsV1;
+
+use super::options::{Display, PercentFormatterOptions};
+
+pub struct FormattedPercent<'l> {
+    pub(crate) value: &'l FixedDecimal,
+    pub(crate) essential: &'l PercentEssentialsV1<'l>,
+    pub(crate) options: &'l PercentFormatterOptions,
+}
+
+impl<'l> Writeable for FormattedPercent<'l> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        match self.options.display {
+            // In the Standard display, we take the
+            Display::Standard => self
+                .essential
+                .positive_pattern
+                .interpolate([self.value])
+                .write_to(sink)?,
+            Display::Approximate => {
+                // Removing the sign from the value
+                let abs_value = match self.value.sign() {
+                    Sign::Negative => self.value.clone().with_sign(Sign::None),
+                    _ => self.value.to_owned(),
+                }
+                .to_string();
+
+                // The approximate sign gets pre-pended
+                let sign = match self.value.sign() {
+                    Sign::Negative => {
+                        String::new()
+                            + &self.essential.approximately_sign
+                            + &self.essential.minus_sign
+                    }
+                    _ => self.essential.approximately_sign.to_string(),
+                };
```

Comment:
Same here. You might need to define something like
```rust
struct Append<W1, W2>(W1, W2);
impl<W1: Writeable, W2: Writeable> Writeable for Append<W1, W2> {
    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
    where
        W: core::fmt::Write + ?Sized,
    {
        self.0.write_to(sink)?;
        self.1.write_to(sink)
    }
}
```

to avoid intermediate allocations.